trieste/acquisition/function.py

```diff
@@ -779,37 +852,70 @@ def prepare_acquisition_function(
+        _partition_bounds = prepare_default_non_dominated_partition_bounds(_pf.front, _reference_pt)
+        function.update(_partition_bounds)  # type: ignore
+        return function
 
-    :param model: The model of the objective function.
-    :param partition_bounds: with shape ([N, D], [N, D]), partitioned non-dominated hypercell
-        bounds for hypervolume improvement calculation
-    :return: The expected_hv_improvement acquisition function modified for objective
-        minimisation. This function will raise :exc:`ValueError` or
-        :exc:`~tf.errors.InvalidArgumentError` if used with a batch size greater than one.
-    """
 
-    def acquisition(x: TensorType) -> TensorType:
+class expected_hv_improvement(AcquisitionFunctionClass):
+    def __init__(self, model: ProbabilisticModel, partition_bounds: tuple[TensorType, TensorType]):
+        r"""
+        expected Hyper-volume (HV) calculating using Eq. 44 of :cite:`yang2019efficient` paper.
+        The expected hypervolume improvement calculation in the non-dominated region
+        can be decomposed into sub-calculations based on each partitioned cell.
+        For easier calculation, this sub-calculation can be reformulated as a combination
+        of two generalized expected improvements, corresponding to Psi (Eq. 44) and Nu (Eq. 45)
+        function calculations, respectively.
+
+        Note:
+        1. Since in Trieste we do not assume the use of a certain non-dominated region partition
+        algorithm, we do not assume the last dimension of the partitioned cell has only one
+        (lower) bound (i.e., minus infinity, which is used in the :cite:`yang2019efficient` paper).
+        This is not as efficient as the original paper, but is applicable to different non-dominated
+        partition algorithm.
+        2. As the Psi and nu function in the original paper are defined for maximization problems,
+        we inverse our minimisation problem (to also be a maximisation), allowing use of the
+        original notation and equations.
+
+        :param model: The model of the objective function.
+        :param partition_bounds: with shape ([N, D], [N, D]), partitioned non-dominated hypercell
+            bounds for hypervolume improvement calculation
+        :return: The expected_hv_improvement acquisition function modified for objective
+            minimisation. This function will raise :exc:`ValueError` or
+            :exc:`~tf.errors.InvalidArgumentError` if used with a batch size greater than one.
+        """
+        self._model = model
+        self._lb_points = tf.Variable(
+            partition_bounds[0], trainable=False, shape=[None, partition_bounds[0].shape[-1]]
+        )
+        self._ub_points = tf.Variable(
+            partition_bounds[1], trainable=False, shape=[None, partition_bounds[1].shape[-1]]
+        )
+        self._cross_index = tf.constant(
+            list(product(*[[0, 1]] * self._lb_points.shape[-1]))
```

Comment:
Not sure I understand what cross_index is for, probably worth explaining in a comment. Also:

```suggestion
            list(product([0, 1] * self._lb_points.shape[-1]))
```

Reply:
Not quite. Product takes a list of arguments, not a single argument that's a list: `product([0,1], [0,1], [0,1]) not product([[0,1],[0,1],[0,1]])`.

Reply:
i stand corrected, thanks